dll/cpl/sysdm/general.c

```diff
@@ -687,21 +698,47 @@ INT_PTR CALLBACK GeneralPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM
             SetWindowLongPtr(GetDlgItem(hwndDlg, IDC_ROSIMG), GWLP_WNDPROC, (LONG_PTR)RosImageProc);
             GetSystemInformation(hwndDlg);
             GetSystemVersion(hwndDlg);
-            GetSystemUptime(hwndDlg);
+            InitSystemUptime(hwndDlg);
             break;
+        }
 
         case WM_DESTROY:
+        {
+            KillTimer(hwndDlg, ID_SYSUPTIME_UPDATE_TIMER);
             HeapFree(GetProcessHeap(), 0, pImgInfo);
+
+            if (hKernel32Vista)
+            {
+                FreeLibrary(hKernel32Vista);
+            }
```

Comment:
Cleaning in the reverse order from initialization.
```suggestion
            if (hKernel32Vista)
                FreeLibrary(hKernel32Vista);

            HeapFree(GetProcessHeap(), 0, pImgInfo);
```

Reply:
Done.